engine/runtime-instrument-id-execution/src/main/java/org/enso/interpreter/instrument/IdExecutionInstrument.java

```diff
@@ -187,6 +191,13 @@ private void onExpressionReturn(Object result, Node node, EventContext context)
         boolean isPanic = result instanceof PanicSentinel;
         UUID nodeId = ((ExpressionNode) node).getId();
         String resultType = Types.getName(result);
+        if (resultType == null && result != null) {
+            // Slow path
+            Object typeResult = typeOfNode.execute(result);
```

Comment:
The whole method is behind `@TruffleBoundary` right now. I suggest to change it and call `typeOfNode.execute` in PE mode. It should be quite fast.